lib/livebook/epmd.ex

```diff
@@ -30,82 +30,50 @@ defmodule Livebook.EPMD do
 
   # Custom EPMD callbacks
 
-  # Custom callback that registers the parent information.
-  # We read this information when trying to connect to the parent.
-  def start_link() do
-    with {:ok, [[node, port]]} <- :init.get_argument(:livebook_parent) do
-      [name, host] = :string.split(node, ~c"@")
-
-      :persistent_term.put(
-        :livebook_parent,
-        {name, host, List.to_atom(node), List.to_integer(port)}
-      )
-    end
-
-    :erl_epmd.start_link()
-  end
-
   # Custom callback to register our current node port.
   def register_node(name, port), do: register_node(name, port, :inet)
 
   def register_node(name, port, family) do
     :persistent_term.put(:livebook_dist_port, port)
-    :erl_epmd.register_node(name, port, family)
+
+    case :erl_epmd.register_node(name, port, family) do
+      {:ok, creation} -> {:ok, creation}
+      {:error, :already_registered} -> {:error, :already_registered}
+      # If registration fails because EPMD is not running, we ignore
+      # that, because we do not rely on EPMD
+      _ -> {:ok, -1}
+    end
   end
 
   # Custom callback that accesses the parent information.
   def port_please(name, host), do: port_please(name, host, :infinity)
 
-  def port_please(name, host, timeout) do
-    case livebook_port(name) do
-      0 -> :erl_epmd.port_please(name, host, timeout)
-      port -> {:port, port, @epmd_dist_version}
-    end
+  def port_please(~c"fly_runtime_" ++ port, _host, _timeout) do
```

Comment:
Should we use a generic name, such as lb_runtime, so we can support other runtimes in the future without changing it here?

Reply:
To be clear, one of the reasons we chose the proxy route is because it also works for k8s, so `lb_runtime` or `proxy_runtime` make sense to me!